README.md

```diff
@@ -198,13 +200,31 @@ rails g parser Date
 **lib/decanter/parsers/date_parser**
 
 ```ruby
-class DateParser < Decanter::ValueParser::Base
-  parser do |name, value, options|
+class DateParser < Decanter::Parser::ValueParser
+  parser do |value, options|
     # your parsing logic here
   end
 end
 ```
 
+
+By inheriting from Decanter::Parser::ValueParser, the assumption is that the value returned from the parser will be the value associated with the provided key. If you need more control over the result, for example, you want a parser that returns multiple key value pairs, you should instead inherit from Decanter::Parser::HashParser. This required that the value returned is a hash. For example:
+
+```ruby
+class KeyValueSplitterParser < Decanter::Parser::HashParser
+  ITEM_DELIM = ','
+  PAIR_DELIM = ':'
+
+  parser do |name, val, options|
+    item_delimiter = options.fetch(:item_delimiter, ITEM_DELIM)
+    pair_delimiter = options.fetch(:pair_delimiter, PAIR_DELIM)
+    val.split(item_delimiter).reduce({}) { |memo, pair| memo.merge( Hash[ *pair.split(pair_delimiter) ] ) }
```

Comment:
@dshvazz if we are putting this example into the Readme, we need the code to be very readable. Can we split this one line into several with descriptive variable names for the return value of each? For example:

```ruby
parser do |name, val, options|
  # val = "color:blue,price:45.31"

  item_delimiter = options.fetch(:item_delimiter, ITEM_DELIM)
  pair_delimiter = options.fetch(:pair_delimiter, PAIR_DELIM)

  pairs = val.split(item_delimiter)
  # pairs = ["color:blue", "price:45.31"]

  hash = {}
  pairs.each do |pair|
    key, value = pair.split(pair_delimiter)
    # key = "color"
    # value = "blue"
    hash[key] = value
  end
  return hash
end
```